Local macOS build fails due to _Atomic enum in switch() after 8381196754bfe348d3561958cc3bc9d04435f714

## Description

When building the SAF examples locally on macOS (Apple Clang), the build fails with errors related to using `_Atomic enum` values inside `switch()` statements.

Example error:

```
error: statement requires expression of integer type ('_Atomic_ARRAY2SH_ARRAY_TYPES' invalid)
    switch(arraySpecs->arrayType)
```

Similar errors appear in several files, e.g.:

* `array2sh_internal.c`
* `beamformer.c`
* `ambi_dec.c`
* `dirass_internal.c`
* etc.

This happens because recent Apple Clang versions treat `_Atomic(T)` as a distinct non-integer type, which is not allowed in a `switch()` expression.
Linux (GCC/Clang) and even macOS runners on GitHub Actions do **not** show this behavior — likely due to older or more permissive compiler versions — so the CI build succeeds while the local macOS build fails.

## Environment

1. OS: MAC
2. Compiler: Clang (llvm)
3. SAF performance library being used: SAF_USE_APPLE_ACCELERATE
5. Other environment details: CMake

## Reproducible Steps

Compile SAF using 

```
Homebrew clang version 17.0.6
Target: arm64-apple-darwin23.5.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm@17/bin
```

## Expected Output

Allow compilation of SAF using clang llvm!

## Actual Output

```
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:235:9: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)                    235 |         switch (arraySpecs->weightType){
      |         ^       ~~~~~~~~~~~~~~~~~~~~~~                                                                              /Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:269:9: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_ARRAY_TYPES' (aka '_Atomic(ARRAY2SH_ARRAY_TYPES)') invalid)
  269 |         switch(arraySpecs->arrayType){
      |         ^      ~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/ar1 error generated.
ray2sh/array2sh_internal.c:271:17: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)                                                                                  271 |                 switch (arraySpecs->weightType){
      |                 ^       ~~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:281:17: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)
  281 |                 switch (arraySpecs->weightType){                                                                          |                 ^       ~~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:415:5: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_ARRAY_TYPES' (aka '_Atomic(ARRAY2SH_ARRAY_TYPES)') invalid)
  415 |     switch(arraySpecs->arrayType){
      |     ^      ~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:420:13: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)                   420 |             switch (arraySpecs->weightType){
      |             ^       ~~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:542:5: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_ARRAY_TYPES' (aka '_Atomic(ARRAY2SH_ARRAY_TYPES)') invalid)
  542 |     switch(arraySpecs->arrayType){
      |     ^      ~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:544:13: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)
  544 |             switch(arraySpecs->weightType){
      |             ^      ~~~~~~~~~~~~~~~~~~~~~~
/Users/neimog/Downloads/Spatial_Audio_Framework/examples/src/array2sh/array2sh_internal.c:574:13: error: statement requires expression of integer type ('_Atomic_ARRAY2SH_WEIGHT_TYPES' (aka '_Atomic(ARRAY2SH_WEIGHT_TYPES)') invalid)
  574 |             switch(arraySpecs->weightType){
      |             ^      ~~~~~~~~~~~~~~~~~~~~~~
```

--- a/examples/src/ambi_bin/ambi_bin.c
+++ b/examples/src/ambi_bin/ambi_bin.c
@@ -292,7 +292,8 @@ void ambi_bin_initCodec
     pData->progressBar0_1 = 0.95f;
     float_complex* decMtx;
     decMtx = calloc1d(HYBRID_BANDS*NUM_EARS*nSH, sizeof(float_complex));
-    switch(pData->method){
+    AMBI_BIN_DECODING_METHODS method = pData->method;
+    switch(method){
         default:
         case DECODING_METHOD_LS:
             getBinauralAmbiDecoderMtx(pars->hrtf_fb, pars->hrir_dirs_deg, pars->N_hrir_dirs, HYBRID_BANDS,

--- a/examples/src/ambi_dec/ambi_dec.c
+++ b/examples/src/ambi_dec/ambi_dec.c
@@ -268,7 +268,8 @@ void ambi_dec_initCodec
     /* calculate loudspeaker decoding matrices */
     for( d=0; d<NUM_DECODERS; d++){
         M_dec_tmp = malloc1d(nLoudspeakers * max_nSH * sizeof(float));
-        switch(pData->dec_method[d]){
+        AMBI_DEC_DECODING_METHODS dec_method = pData->dec_method[d];
+        switch(dec_method){
             case DECODING_METHOD_SAD:
                 getLoudspeakerDecoderMtx((float*)loudpkrs_dirs_deg_local, nLoudspeakers, LOUDSPEAKER_DECODER_SAD, masterOrder, 0, M_dec_tmp);
                 break;

--- a/examples/src/array2sh/array2sh_internal.c
+++ b/examples/src/array2sh/array2sh_internal.c
@@ -136,9 +136,11 @@ void array2sh_calculate_sht_matrix
         /* Compute modal responses */
         free(pData->bN);
         pData->bN = malloc1d((HYBRID_BANDS)*(order+1)*sizeof(double_complex));
-        switch(arraySpecs->arrayType){
-            case ARRAY_CYLINDRICAL:
-                switch (arraySpecs->weightType){
+        ARRAY2SH_ARRAY_TYPES arrayType = arraySpecs->arrayType;
+        switch(arrayType){
+            case ARRAY_CYLINDRICAL: {
+                ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+                switch (weightType){
                     case WEIGHT_RIGID_OMNI:   cylModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_RIGID, pData->bN); break;
                     case WEIGHT_RIGID_CARD:   saf_print_error("weightType is not supported"); break;
                     case WEIGHT_RIGID_DIPOLE: saf_print_error("weightType is not supported"); break;
@@ -147,8 +149,10 @@ void array2sh_calculate_sht_matrix
                     case WEIGHT_OPEN_DIPOLE:  saf_print_error("weightType is not supported"); break;
                 }
                 break;
-            case ARRAY_SPHERICAL:
-                switch (arraySpecs->weightType){
+            }
+            case ARRAY_SPHERICAL: {
+                ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+                switch (weightType){
                     case WEIGHT_OPEN_OMNI:   sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN, 1.0, pData->bN); break;
                     case WEIGHT_OPEN_CARD:   sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.5, pData->bN); break;
                     case WEIGHT_OPEN_DIPOLE: sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.0, pData->bN); break;
@@ -171,6 +175,7 @@ void array2sh_calculate_sht_matrix
                         break;
                 }
                 break;
+            }
         }
         
         for(band=0; band<HYBRID_BANDS; band++)
@@ -232,7 +237,8 @@ void array2sh_calculate_sht_matrix
         double_complex Hs[HYBRID_BANDS][MAX_SH_ORDER+1];
         
         /* find suitable cut-off frequencies */
-        switch (arraySpecs->weightType){
+        ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+        switch (weightType){
             case WEIGHT_OPEN_OMNI:   sphArrayNoiseThreshold(order, arraySpecs->Q, arraySpecs->r, pData->c, ARRAY_CONSTRUCTION_OPEN, 1.0, pData->regPar, f_lim); break;
             case WEIGHT_OPEN_CARD:   sphArrayNoiseThreshold(order, arraySpecs->Q, arraySpecs->r, pData->c, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.5, pData->regPar, f_lim); break;
             case WEIGHT_OPEN_DIPOLE: sphArrayNoiseThreshold(order, arraySpecs->Q, arraySpecs->r, pData->c, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.0, pData->regPar, f_lim); break;
@@ -266,9 +272,11 @@ void array2sh_calculate_sht_matrix
         /* compute inverse radial response */ 
         free(pData->bN);
         pData->bN = malloc1d((HYBRID_BANDS)*(order+1)*sizeof(double_complex));
-        switch(arraySpecs->arrayType){
-            case ARRAY_CYLINDRICAL:
-                switch (arraySpecs->weightType){
+        ARRAY2SH_ARRAY_TYPES arrayType = arraySpecs->arrayType;
+        switch(arrayType){
+            case ARRAY_CYLINDRICAL:{
+                ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->arrayType;
+                switch (weightType){
                     case WEIGHT_RIGID_OMNI:   cylModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_RIGID, pData->bN); break;
                     case WEIGHT_RIGID_CARD:   /* not supported */ break;
                     case WEIGHT_RIGID_DIPOLE: /* not supported */ break;
@@ -277,8 +285,10 @@ void array2sh_calculate_sht_matrix
                     case WEIGHT_OPEN_DIPOLE:  /* not supported */ break;
                 }
                 break;
-            case ARRAY_SPHERICAL:
-                switch (arraySpecs->weightType){
+            }
+            case ARRAY_SPHERICAL:{
+                ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->arrayType;
+                switch (weightType){
                     case WEIGHT_OPEN_OMNI:   sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN, 1.0, pData->bN); break;
                     case WEIGHT_OPEN_CARD:   sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.5, pData->bN); break;
                     case WEIGHT_OPEN_DIPOLE: sphModalCoeffs(order, kr, HYBRID_BANDS, ARRAY_CONSTRUCTION_OPEN_DIRECTIONAL, 0.0, pData->bN); break;
@@ -301,6 +311,7 @@ void array2sh_calculate_sht_matrix
                         break;
                 }
                 break;
+            }
         }
         
         /* direct inverse (only required for GUI) */
@@ -412,12 +423,14 @@ void array2sh_apply_diff_EQ(void* const hA2sh)
     }
     
     /* Get theoretical diffuse coherence matrix */
-    switch(arraySpecs->arrayType){
+    ARRAY2SH_ARRAY_TYPES arrayType = arraySpecs->arrayType;
+    switch(arrayType){
         case ARRAY_CYLINDRICAL:
             return; /* Unsupported */
             break;
-        case ARRAY_SPHERICAL:
-            switch (arraySpecs->weightType){
+        case ARRAY_SPHERICAL:{
+            ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+            switch (weightType){
                 case WEIGHT_RIGID_OMNI: /* Does not handle the case where kr != kR ! */
                     sphDiffCohMtxTheory(array_order, (float*)sensorCoords_rad_local, arraySpecs->Q, ARRAY_CONSTRUCTION_RIGID, 1.0, kr, HYBRID_BANDS, dM_diffcoh);
                     break;
@@ -438,6 +451,7 @@ void array2sh_apply_diff_EQ(void* const hA2sh)
                     break;
             }
             break;
+        }
     }
 
     /* determine band index for the spatial aliasing limit */
@@ -539,9 +553,11 @@ void array2sh_evaluateSHTfilters(void* hA2sh)
         kR[band] = 2.0*SAF_PId*(pData->freqVector[band])*(arraySpecs->R)/pData->c;
     }
     H_array = malloc1d((HYBRID_BANDS) * (arraySpecs->Q) * 812*sizeof(float_complex));
-    switch(arraySpecs->arrayType){
-        case ARRAY_SPHERICAL:
-            switch(arraySpecs->weightType){
+    ARRAY2SH_ARRAY_TYPES arrayType = arraySpecs->arrayType;
+    switch(arrayType){
+        case ARRAY_SPHERICAL:{
+            ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+            switch(weightType){
                 default:
                 case WEIGHT_RIGID_OMNI:
                     simulateSphArray(simOrder, kr, kR, HYBRID_BANDS, (float*)arraySpecs->sensorCoords_rad, arraySpecs->Q,
@@ -569,9 +585,10 @@ void array2sh_evaluateSHTfilters(void* hA2sh)
                     break;
             }
             break;
-            
-        case ARRAY_CYLINDRICAL:
-            switch(arraySpecs->weightType){
+        }
+        case ARRAY_CYLINDRICAL:{
+            ARRAY2SH_WEIGHT_TYPES weightType = arraySpecs->weightType;
+            switch(weightType){
                 default:
                 case WEIGHT_RIGID_OMNI:
                 case WEIGHT_RIGID_CARD:
@@ -585,6 +602,7 @@ void array2sh_evaluateSHTfilters(void* hA2sh)
                     break;
             }
             break;
+        }
     }
     
     strcpy(pData->progressBarText,"Evaluating encoding performance");

--- a/examples/src/beamformer/beamformer.c
+++ b/examples/src/beamformer/beamformer.c
@@ -141,7 +141,8 @@ void beamformer_process
         for(bi=0; bi<nBeams; bi++){
             if(pData->recalc_beamWeights[bi]){
                 memset(pData->beamWeights[bi], 0, MAX_NUM_SH_SIGNALS*sizeof(float));
-                switch(pData->beamType){
+                STATIC_BEAM_TYPES beamType = pData->beamType;
+                switch(beamType){
                     case STATIC_BEAM_TYPE_CARDIOID: beamWeightsCardioid2Spherical(beamOrder, c_n); break;
                     case STATIC_BEAM_TYPE_HYPERCARDIOID: beamWeightsHypercardioid2Spherical(beamOrder, c_n); break;
                     case STATIC_BEAM_TYPE_MAX_EV: beamWeightsMaxEV(beamOrder, c_n); break;

--- a/examples/src/dirass/dirass.c
+++ b/examples/src/dirass/dirass.c
@@ -665,14 +665,16 @@ int dirass_getPmap(void* const hDir, float** grid_dirs, float** pmap, int* nDirs
         (*pmap) = pData->pmap_grid[pData->dispSlotIdx-1 < 0 ? NUM_DISP_SLOTS-1 : pData->dispSlotIdx-1];
         (*nDirs) = pars->interp_nDirs;
         (*pmapWidth) = pData->dispWidth;
-        switch(pData->HFOVoption){
+        HFOV_OPTIONS HFOVoption = pData->HFOVoption;
+        switch(HFOVoption){
             default:
             case HFOV_360: (*hfov) = 360; break;
             case HFOV_180: (*hfov) = 180; break;
             case HFOV_90:  (*hfov) = 90;  break;
             case HFOV_60:  (*hfov) = 60;  break;
         }
-        switch(pData->aspectRatioOption){
+        ASPECT_RATIO_OPTIONS  aspectRatioOption = pData->aspectRatioOption;
+        switch(aspectRatioOption){
             default:
             case ASPECT_RATIO_2_1:  (*aspectRatio) = 2.0f; break;
             case ASPECT_RATIO_16_9: (*aspectRatio) = 16.0f/9.0f; break;

--- a/examples/src/dirass/dirass_internal.c
+++ b/examples/src/dirass/dirass_internal.c
@@ -66,7 +66,8 @@ void dirass_initAna(void* const hDir)
     pData->progressBar0_1 = 0.4f;
     
     /* Determine scanning grid */
-    switch(pData->gridOption){
+    DIRASS_GRID_OPTIONS gridOption = pData->gridOption;
+    switch(gridOption){
         case T_DESIGN_3:           /* 6 points */
             td_degree = 3;
             pars->grid_dirs_deg = (float*)__HANDLES_Tdesign_dirs_deg[td_degree-1];
@@ -129,14 +130,17 @@ void dirass_initAna(void* const hDir)
     }
     
     /* generate interpolation table for current display config */
-    switch(pData->HFOVoption){
+    HFOV_OPTIONS HFOVoption = pData->HFOVoption;
+    switch(HFOVoption){
         default:
         case HFOV_360: hfov = 360.0f; break;
         case HFOV_180: hfov = 180.0f; break;
         case HFOV_90:  hfov = 90.0f;  break;
         case HFOV_60:  hfov = 60.0f;  break;
     }
-    switch(pData->aspectRatioOption){
+
+    ASPECT_RATIO_OPTIONS aspectRatioOption = pData->aspectRatioOption;
+    switch(aspectRatioOption){
         default:
         case ASPECT_RATIO_2_1:  aspectRatio = 2.0f; break;
         case ASPECT_RATIO_16_9: aspectRatio = 16.0f/9.0f; break;
@@ -174,7 +178,8 @@ void dirass_initAna(void* const hDir)
     A_xyz = malloc1d(nSH_order*nSH_sec*3*sizeof(float_complex));
     c_n = malloc1d((order_sec+1)*sizeof(float));
     computeVelCoeffsMtx(order_sec, A_xyz);
-    switch(pData->beamType){
+    STATIC_BEAM_TYPES beamType = pData->beamType;
+    switch(beamType){
         case STATIC_BEAM_TYPE_CARDIOID: beamWeightsCardioid2Spherical(order_sec, c_n); break;
         case STATIC_BEAM_TYPE_HYPERCARDIOID: beamWeightsHypercardioid2Spherical(order_sec, c_n); break;
         case STATIC_BEAM_TYPE_MAX_EV: beamWeightsMaxEV(order_sec, c_n); break;
@@ -192,7 +197,7 @@ void dirass_initAna(void* const hDir)
 
     /* get regular beamforming weights */
     c_n = malloc1d((order+1)*sizeof(float));
-    switch(pData->beamType){
+    switch(beamType){
         case STATIC_BEAM_TYPE_CARDIOID: beamWeightsCardioid2Spherical(order, c_n); break;
         case STATIC_BEAM_TYPE_HYPERCARDIOID: beamWeightsHypercardioid2Spherical(order, c_n); break;
         case STATIC_BEAM_TYPE_MAX_EV: beamWeightsMaxEV(order, c_n); break;
@@ -206,7 +211,7 @@ void dirass_initAna(void* const hDir)
  
     /* beamforming weights for upscaled */
     c_n = malloc1d((order_up+1)*sizeof(float)); 
-    switch(pData->beamType){
+    switch(beamType){
         case STATIC_BEAM_TYPE_CARDIOID: beamWeightsCardioid2Spherical(order_up, c_n); break;
         case STATIC_BEAM_TYPE_HYPERCARDIOID: beamWeightsHypercardioid2Spherical(order_up, c_n); break;
         case STATIC_BEAM_TYPE_MAX_EV: beamWeightsMaxEV(order_up, c_n); break;

--- a/examples/src/pitch_shifter/pitch_shifter.c
+++ b/examples/src/pitch_shifter/pitch_shifter.c
@@ -128,7 +128,9 @@ void pitch_shifter_initCodec
         smb_pitchShift_destroy(&(pData->hSmb));
 
     /* Config */
-    switch(pData->osamp_option){
+
+    PITCH_SHIFTER_OSAMP_OPTIONS osamp_option = pData->osamp_option;
+    switch(osamp_option){
         default:
             /* fall through */
         case PITCH_SHIFTER_OSAMP_2:  osamp = 2; break;
@@ -137,7 +139,8 @@ void pitch_shifter_initCodec
         case PITCH_SHIFTER_OSAMP_16: osamp = 16; break;
         case PITCH_SHIFTER_OSAMP_32: osamp = 32; break;
     }
-    switch(pData->fftsize_option){
+    PITCH_SHIFTER_FFTSIZE_OPTIONS fftsize_option = pData->fftsize_option;
+    switch(fftsize_option){
         default:
             /* fall through */
         case PITCH_SHIFTER_FFTSIZE_512:   fftSize = 512; break;

--- a/examples/src/powermap/powermap.c
+++ b/examples/src/powermap/powermap.c
@@ -728,13 +728,15 @@ int powermap_getPmap(void* const hPm, float** grid_dirs, float** pmap, int* nDir
         (*pmap) = pData->pmap_grid[pData->dispSlotIdx-1 < 0 ? NUM_DISP_SLOTS-1 : pData->dispSlotIdx-1];
         (*nDirs) = pars->interp_nDirs;
         (*pmapWidth) = pData->dispWidth;
-        switch(pData->HFOVoption){
+        HFOV_OPTIONS HFOVoption = pData->HFOVoption;
+        switch(HFOVoption){
             default:
             case HFOV_360:
                 (*hfov) = 360;
                 break;
         }
-        switch(pData->aspectRatioOption){
+        ASPECT_RATIO_OPTIONS aspectRatioOption = pData->aspectRatioOption;
+        switch(aspectRatioOption){
             default:
             case ASPECT_RATIO_2_1:
                 (*aspectRatio) = 2;

--- a/examples/src/powermap/powermap_internal.c
+++ b/examples/src/powermap/powermap_internal.c
@@ -75,11 +75,13 @@ void powermap_initAna(void* const hPm)
 
     if(pData->isFirstInit){
         /* generate interpolation table for current display settings */
-        switch(pData->HFOVoption){
+        HFOV_OPTIONS HFOVoption = pData->HFOVoption;
+        switch(HFOVoption){
             default:
             case HFOV_360: hfov = 360.0f; break;
         }
-        switch(pData->aspectRatioOption){
+        ASPECT_RATIO_OPTIONS aspectRatioOption = pData->aspectRatioOption;
+        switch(aspectRatioOption){
             default:
             case ASPECT_RATIO_2_1: aspectRatio = 2.0f; break;
         }

